packages/video_player/video_player/lib/video_player.dart

```diff
@@ -728,20 +728,36 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
   ///
   /// If no [closedCaptionFile] was specified, this will always return an empty
   /// [Caption].
+
   Caption _getCaptionAt(Duration position) {
     if (_closedCaptionFile == null) {
       return Caption.none;
     }
 
     final Duration delayedPosition = position + value.captionOffset;
-    // TODO(johnsonmh): This would be more efficient as a binary search.
-    for (final Caption caption in _closedCaptionFile!.captions) {
-      if (caption.start <= delayedPosition && caption.end >= delayedPosition) {
-        return caption;
+
+    final List<Caption> captions = _closedCaptionFile!.captions;
+
+    int left = 0;
+    int right = captions.length - 1;
+
+    while (left <= right) {
```

Comment:
https://github.com/flutter/plugins/pull/2487#discussion_r376564502

> Not a big deal either way, but for what it's worth I think you could trivially import this. https://api.flutter.dev/flutter/package-collection_collection/binarySearch.html

> Yes, it wouldn't be too hard, although we would have to either sort the captions at initialization, or make it clear that we assume a duration sorted list of captions (neither of which we currently do).

Reply:
I implemented all the mentioned changes. However, to use the collection function, I had to convert to a list, so I cached it to avoid re-mapping on each check.

Reply:
I think what @jmagman is suggesting is that there is a package should be used to do the binary search.